genanki/__init__.py

```diff
@@ -151,29 +152,88 @@ def to_json(self, now_ts, deck_id):
     }
 
 
+class CardOptions:
+  def __init__(self, stage=0, status=0, due=0, interval=0,
+               ease_factor=0, reps_left_til_grad=0):
+    """SRS learning stage.
+    0 = new, 1 = learning, 2 = review."""
+    self.type = stage
+    """SRS queue status modifiers.
+    0 = normal, 1 = suspended, 2 = user buried, 3 = scheduler buried"""
+    self.status = status
+    """Behavior depends on learning stage of note.
+       new: unused.
+       learning: due time as integer seconds since Unix epoch.
+       review: integer days relative to deck creation timestamp."""
+    self.due = due
+    """Time between next review and the one following.
+    Positive values are in days, negative in seconds."""
+    self.ivl = interval
+    """Integer 'ease' factor used by SRS algorithm.
+    Example: 2500 = 250%."""
+    self.factor = ease_factor
+    """Repititions remaining until graduation from the learning stage.
+    Unused during other SRS stages."""
+    self.left = reps_left_til_grad
+
+  @property
+  def stage(self):
+    return self.type
+  @stage.setter
+  def stage(self, value):
+    self.type = value
+  @property
+  def interval(self):
+    return self.ivl
+  @interval.setter
+  def interval(self, value):
+    self.ivl = value
+  @property
+  def ease_factor(self):
+    return self.factor
+  @ease_factor.setter
+  def ease_factor(self, value):
+    self.factor = value
+  @property
+  def reps_left_til_grad(self):
+    return self.left
+  @reps_left_til_grad.setter
+  def reps_left_til_grad(self, value):
+    self.left = value
+
+
 class Card:
-  def __init__(self, ord):
+  def __init__(self, ord, options=None):
```

Comment:
With this `options` object, you can't directly set properties on the `Card`. For example, you can't do `my_card.type = 1`, you have to do `my_card.options.type = 1`, which seems like an unnecessary level of indirection and doesn't mesh with how other objects (`Note`s and `Deck`s) work. Can you make these direct properties of `Card` and get rid of the `CardOptions` class?

To achieve what you're trying to do with `CardOptions`, you can write a helper function (in your own code) like
```
def set_card_attributes(card, attrs):
    for a in attr:
        setattr(card, a, attrs[a])
```
and then write code like
```
my_attrs = {'type': 2, 'due': 1234}
for note in notes:
    for card in note.cards:
        set_card_attributes(card, my_attrs)
```